payjoin/tests/integration.rs

```diff
@@ -281,6 +281,163 @@ mod integration {
             Ok(())
         }
 
+        #[tokio::test]
+        async fn test_enable_output_substitution_v1_to_v2() -> Result<(), BoxSendSyncError> {
+            init_tracing();
+            let mut services = TestServices::initialize().await?;
+            let result = tokio::select!(
+            err = services.take_ohttp_relay_handle() => panic!("Ohttp relay exited early: {:?}", err),
+            err = services.take_directory_handle() => panic!("Directory server exited early: {:?}", err),
+            res = do_enable_output_substitution_v1_to_v2(&services) => res
+            );
+
+            assert!(result.is_ok(), "v2 send receive failed: {:#?}", result.unwrap_err());
+
+            async fn do_enable_output_substitution_v1_to_v2(
+                services: &TestServices,
+            ) -> Result<(), BoxError> {
+                let (_bitcoind, sender, receiver) = init_bitcoind_sender_receiver(None, None)?;
+                let agent = services.http_agent();
+                services.wait_for_services_ready().await?;
+                let directory = services.directory_url();
+                let ohttp_keys = services.fetch_ohttp_keys().await?;
+                let recv_persister = NoopSessionPersister::default();
+                let send_persister = NoopSessionPersister::default();
+                let address = receiver.get_new_address(None, None)?.assume_checked();
+                let mut session =
+                    Receiver::create_session(address, directory.clone(), ohttp_keys.clone(), None)
+                        .save(&recv_persister)?;
+
+                // **********************
+                // Inside the V1 Sender:
+                // Create a funded PSBT (not broadcasted) to address with amount given in the pj_uri
+                assert_eq!(
+                    session.pj_uri().extras.output_substitution(),
+                    OutputSubstitution::Disabled
+                );
+                let enabled_output_substitution =
+                    session.pj_uri().to_string().replace("pjos=0", "pjos=1");
+                let pj_uri = Uri::from_str(&enabled_output_substitution)
+                    .map_err(|e| e.to_string())?
+                    .assume_checked()
+                    .check_pj_supported()
+                    .map_err(|e| e.to_string())?;
+                // We need to check that output substitution has been forced anabled
```

Comment:
typo:
```suggestion
                // We need to check that output substitution is enabled
```
(or remove comment)